### DISCUSSION TITLE
Granular access

### DISCUSSION THREAD

**Discussion by Vladimir-Va** [nocodb/nocodb#399] [Features]
**Columns Users and Roles**
Add columns User and Roles field types (O2M).

**[Сheckbox"Fill" in filter row dialog.](https://github.com/nocodb/nocodb/discussions/398)**
If we setup filters with Column_name is_equal Values it is logical that when adding a new row, we want to get the value in the columns equal to the values ​​set in the filters.

**Сheckbox "Predicate" in filter row dialog.**
Predicate Mandatory Filters that only the Owner can configure.

**Global Variables get login in user %USER% and %ROLE%**
For Granular access case set two filters:
_"Users column is equal %USER%", Сheckboxes "Fill"=1 and "Predicate"=1
"Roles column is equal %ROLE%", Сheckboxes "Fill"=1 and "Predicate"=1_

**Comment by o1lab** [comment]
Can be on the longer roadmap. Good idea. Need a bit more thinking.

**Reply by Vladimir-Va** [reply]
Example
"Fill" checkbox visible if operation = is equal
![nocodb_filters](https://user-images.githubusercontent.com/52074398/127852173-11b00fd7-5eed-464a-96d9-e938eafd932c.png)
